src/com/maddyhome/idea/vim/action/motion/updown/MotionDownAction.java

```diff
@@ -36,16 +41,40 @@ public MotionDownAction() {
   }
 
   private static class Handler extends MotionEditorActionHandler {
-    public int getOffset(@NotNull Editor editor, DataContext context, int count, int rawCount, Argument argument) {
-      return VimPlugin.getMotion().moveCaretVertical(editor, count);
+    public Handler() {
+      super(true);
     }
 
-    protected void preMove(@NotNull Editor editor, DataContext context, Command cmd) {
-      col = EditorData.getLastColumn(editor);
+    @Override
+    public int getOffset(@NotNull Editor editor, @NotNull Caret caret, @NotNull DataContext context, int count,
```

Comment:
The down motion doesn't work as expected in several cases when you move around in a file with multiple carets.

For example, given this file:

```
<caret>foo
<caret>bar
baz
```

when I type `lj`, I expect the result to be:

```
foo
b<caret>ar
b<caret>az
```

but the actual result is:

```
foo
<caret>bar
b<caret>az
```